File: C++/special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int lo = 0, hi = n - 1;
        while (lo <= hi) {
            int i = (lo + hi) / 2;
            int x = n - i;
            if (nums[i] >= x) {
                if (i == 0 || nums[i-1] < x) {
                    return x;
                } else {
                    hi = i - 1;
                }
            } else {
                lo = i + 1;
            }
        }
        return -1;
    }
};

File: Go/special-array-with-x-elements-greater-than-or-equal-x.go
func specialArray(nums []int) int {
    for x := 0; x <= len(nums); x++ {
        count := 0
        for _, n := range nums {
            if n >= x {
                count++
            }
        }
        if count == x {
            return x
        }
    }
    return -1
}